agent/reporter.go

```diff
@@ -13,12 +13,20 @@ import (
 // concurrently, so implementations should be goroutine-safe.
 type StatusReporter interface {
 	UpdateTaskStatus(ctx context.Context, taskID string, status *api.TaskStatus) error
+	ID() string
 }
 
-type statusReporterFunc func(ctx context.Context, taskID string, status *api.TaskStatus) error
+type statusReporterFunc struct {
+	function func(ctx context.Context, taskID string, status *api.TaskStatus) error
```

Comment:
This is pretty ugly.  :|  But the alternative is to always report "anonymous function".  In reality, we only use this (not in tests) in one place, and that is an anonymous function for reporting on a single task.